nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/util/TgzParser.java

```diff
@@ -33,32 +29,25 @@
  */
 @Named
 @Singleton
-public class TgzParser
-{
+public class TgzParser {
+
   private static final String CHART_NAME = "Chart.yaml";
 
-  @Nullable
   public InputStream getChartFromInputStream(final InputStream is) throws IOException {
     try (GzipCompressorInputStream gzis = new GzipCompressorInputStream(is)) {
       try (TarArchiveInputStream tais = new TarArchiveInputStream(gzis)) {
-        ArchiveEntry currentEntry = tais.getNextEntry();
-
-        while (currentEntry != null) {
-          if (currentEntry.getName().contains(CHART_NAME)) {
+        ArchiveEntry currentEntry;
+        while ((currentEntry = tais.getNextEntry()) != null) {
+          String entryName = currentEntry.getName();
+          if (entryName.contains(CHART_NAME) &&
```

Comment:
entryName.endswith ?

Reply:
only two possible names you need "Chart.yaml" and "./Chart.yaml"

Reply:
Yes, you are right. endsWith will be better. 
https://github.com/sonatype-nexus-community/nexus-repository-helm/pull/99/commits/0b7d5ad18bc2fa72e47784a81af8829b4e887ee7

Possible, valid names
chart-name/Chart.yaml

mysql/Chart.yaml
mongodb/Chart.yaml